src/arrayMethodSort.js

```diff
@@ -4,8 +4,37 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    let defaultCompare;
+
+    if (typeof compareFunction === 'function') {
+      defaultCompare = compareFunction;
+    } else {
+      defaultCompare = function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA[0] === strB[0]) {
+          return strA.localeCompare(strB);
+        } else {
+          return strA[0].charCodeAt(0) - strB[0].charCodeAt(0);
+        }
```

Comment:
The current implementation of the default comparison function only compares the first character when they are the same. This can lead to incorrect sorting if the remaining parts of the strings differ. The `localeCompare` method should be sufficient for comparing the whole strings.